interview-questions.md

```diff
@@ -7,60 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer:git is a version control system that track what ever modifications and changes you made on your source code history.Git is a hidden folder in your local machine While github is remote based code hosting managment system of git repository.
 
-  Researched answer:
+  Researched answer:Git is version control system that controls and keep track of what ever changes you made in your directory localy where as github is hosting a copy of your local directory and allows you access to share your repository with other developers.
 
 
 
 2. What types of evaluations will return a Boolean value?
 
-  Your answer:
+  Your answer:numeric comparisons always return boolean values.when we compare numbers using logical operators like
+  >,<.===,==,!= etc
 
-  Researched answer:
+  Researched answer:Logical operators when used in logical comparisons usually return boolean values but some time when the && and || logical operators are used with non logical values ,they may return non boolean values.example console.log(0||null) will return none boolean value the null.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer:Index is a numerical value that shows the position of a specific element in an array or a particular character in a string.Index is always numerical value that can be greater or equal to zero where as value can be number,string,boolean ,array or function etc
 
-  Researched answer:
+  Researched answer:Index is a numerical value that indicates the position of an element in an array.It is a reference point when you want to update your array.
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer:Built-in method is a function developed by javascript developers and embeded to the javascript engine
 
-  Researched answer:
+  Researched answer:There are many built in methods in javascript,when the stored data in an object is a function,it is called a method.Method is what an object does while property is what an object has.
```

Comment:
Yes, a method is a function that belongs to and manipulates the object it is inside of.